### ISSUE TITLE
Code-based configuration

### ISSUE THREAD

**Issue by andrewdavey** [andrewdavey/cassette#17]
It seems that people will want control over what asset processing occurs (Sass, Less, JSLint, JSHint, CoffeeScript, etc) . Configuring all this in web.config will be awful. A better approach will be a code-based configuration system.

On application start-up an asset pipeline initializer should be called. This will define the pipeline stages. The nuget package can put a sensible default into the project.

I'm imagining something like:

```
AssetPipelines.Scripts = new Pipeline[] {
    JSLint(),
    CompileCoffeeScript(),
    Concatenate(),
    MSAjaxMinify()
};
```

This will enable people to insert their own asset pipeline processors as well by implementing a simple interface.

For per-module configuration, a `Conditional` pipeline stage wrapper could be used:

```
new Conditional(module => module.Path != "scripts/third-party", JSLint())
```

**Comment by sfmskywalker** [comment]
I vote for it. Not that I require it anymore, but would there also be a CssMinify() pipeline in case I would want to skip that step? :)

**Comment by andrewdavey** [comment]
Yes, there would be a separate pipeline for each asset type (scripts, stylesheets, html templates). So each can be configured any way you like.

**Comment by andrewdavey** [comment]
Implemented in 0.6.
